### DISCUSSION TITLE
New task logs appended to deleted task logs for scheduled runs

### DISCUSSION THREAD

**Discussion by jyotsa09** [apache/airflow#22148] [General]
### Apache Airflow version

main (development)

### What happened

Deleted task logs appended to new logs for scheduled runs

### What you expected to happen

Only new logs should display

### How to reproduce

1. Triggered a scheduled run
2. Once completed delete the dag from UI
3. Triggered the same dag
4. Then check logs

### Operating System

Debian buster

### Deployment

Astronomer

### Deployment details

astro dev start

### Anything else

Dag - https://gist.github.com/jyotsa09/4544ff3b0ccdb7e4caa53741b877e38d#file-gistfile1-txt
Logs - https://gist.github.com/jyotsa09/c6b714dbabcdff7d08694af140768d42#file-gistfile1-txt

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by potiuk** [comment]
When you delete Dag from the UI, it clearly explains that logs are not going to be deleted. Please read carefully the information displayed there.  

This is deliberate decision to keep the logs (for audit and traceabliity purpose) if you want to delete the logs you should delete them manually. Airflow does not manage the logs other than creating them when needed.
